### PR TITLE
Remove unused JVMPI options

### PR DESCRIPTION
**Pull request by alexey-anufriev**
After the cleanup done in https://github.com/eclipse/openj9/pull/10862 these options can be removed

### CODE REVIEW

File: compiler/control/OMROptions.hpp [hunk 1]
```diff
@@ -234,7 +234,6 @@ enum TR_CompilationOptions
    TR_EnableYieldVMAccess                 = 0x02000000 + 4,
    TR_DisableNoVMAccess                   = 0x04000000 + 4,
    TR_DisableStoreSinking                 = 0x08000000 + 4,
-   TR_EnableJVMPILineNumbers              = 0x10000000 + 4,
```
Comment (0xdaryl) [inline compiler/control/OMROptions.hpp:237]: Please don't delete this outright.  Instead, replace it with an "Available" comment to encourage re-use, like in:
https://github.com/eclipse/omr/blob/b406ab2841765a3a182850d8884a9c0297845e00/compiler/control/OMROptions.hpp#L104

File: compiler/control/OMROptions.hpp [hunk 2]
```diff
@@ -333,7 +332,6 @@ enum TR_CompilationOptions
    TR_EnableLowerCompilationLimitsDecisionMaking
                                           = 0x00000020 + 8,
    TR_DisableDirectToJNI                  = 0x00000040 + 8,
-   TR_OldJVMPI                            = 0x00000080 + 8,
```
Comment (0xdaryl) [inline compiler/control/OMROptions.hpp:336]: Same comment as above on deleting outright.

**Review by 0xdaryl — CHANGES_REQUESTED** [review]

File: compiler/control/OMROptions.hpp (hunk 2 above)
Comment (alexey-anufriev) [reply compiler/control/OMROptions.hpp:336]: Fixed

File: compiler/control/OMROptions.hpp (hunk 1 above)
Comment (alexey-anufriev) [reply compiler/control/OMROptions.hpp:237]: Fixed

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**